File: Java/src/World/body/animal/Human.java
package World.body.animal;

import GUI.Color_obj;
import World.Point;
import World.World;
import World.body.Animal;
import World.body.Body;

import java.awt.event.KeyEvent;
import java.io.Serializable;

public class Human extends Animal implements Serializable {
    private static final long serialVersionUID = -1534372028747757553L;

    private int additional_power = 0;
    int age_when_used_special_skill = -10;

    public Human() {
    }

    public Human(Point point_location, World world) {
        super(point_location, world);
        setName("Human");
        setColor(Color_obj.HUMAN);
        setInitiative(4);
        setPower(5);
    }

    @Override
    protected boolean repel_attack(Body attacker) {
        return false;
    }

    @Override
    protected void action() {
        switch(getWorld().getOrder()){
            case KeyEvent.VK_UP:
                setLast_position(getPoint_location());
                setPoint_location(new Point(getPoint_location().getX(), getPoint_location().getY()-1));
                break;
            case KeyEvent.VK_DOWN:
                setLast_position(getPoint_location());
                setPoint_location(new Point(getPoint_location().getX(), getPoint_location().getY()+1));
                break;
            case KeyEvent.VK_LEFT:
                setLast_position(getPoint_location());
                setPoint_location(new Point(getPoint_location().getX()-1, getPoint_location().getY()));
                break;
            case KeyEvent.VK_RIGHT:
                setLast_position(getPoint_location());
                setPoint_location(new Point(getPoint_location().getX()+1, getPoint_location().getY()));
                break;
        }

    }

    public boolean can_use_ability() {
        if(age_when_used_special_skill+10<=getAge()){
            return true;
        }
        return false;
    }

    public void setAdditionalPower(int i) {
        additional_power=i;
    }

    public int getAdditionalPower() {
        return additional_power;
    }

    public void setAgeWhenUsedSpecialSkill(int age) {
        age_when_used_special_skill=age;
    }
}

File: Python/World/Body/Animal/human.py
from GUI.colors import Colors
from World.Body.animal import Animal
from World.point import Point


class Human(Animal):
    additional_power = 0
    age_when_used_special_skill = -10

    def __init__(self, point_location, world):
        super().__init__(point_location, world)
        self.set_name("Human")
        self.set_color(Colors.HUMAN)
        self.set_initiative(4)
        self.set_power(5)

    def repel_attack(self, attacker):
        return False

    def action(self):
        order = self.get_world().get_order()
        if self.get_world().get_order() == 'Up':
            self.set_last_position(self.get_point_location())
            self.set_point_location(Point(self.get_point_location().x, self.get_point_location().y - 1))
        elif order == "Down":
            self.set_last_position(self.get_point_location())
            self.set_point_location(Point(self.get_point_location().x, self.get_point_location().y + 1))
        elif order == "Left":
            self.set_last_position(self.get_point_location())
            self.set_point_location(Point(self.get_point_location().x - 1, self.get_point_location().y))
        elif order == "Right":
            self.set_last_position(self.get_point_location())
            self.set_point_location(Point(self.get_point_location().x + 1, self.get_point_location().y))

    def can_use_ability(self):
        if self.age_when_used_special_skill + 10 <= self.get_age():
            return True
        return False

    def set_additional_power(self, i):
        self.additional_power = i

    def get_additional_power(self):
        return self.additional_power

    def set_age_when_used_special_skill(self, age):
        self.age_when_used_special_skill = age
